File: LLM-Translations/GPT-4/codenet/Python/Go/s763097776.go
package main

import (
	"fmt"
	"sort"
)

func main() {
	var n int
	fmt.Scan(&n)

	p := make([]int, n)
	p_copy := make([]int, n)

	for i := 0; i < n; i++ {
		fmt.Scan(&p[i])
		p_copy[i] = p[i]
	}

	sort.Ints(p)

	count := 0
	for i := 0; i < n; i++ {
		if p[i] != p_copy[i] {
			count += 1
		}
	}

	if count == 0 || count == 2 {
		fmt.Println("YES")
	} else {
		fmt.Println("NO")
	}
}
// 

File: LLM-Translations/GPT-4/codenet/Python/C/s763097776.c
#include <stdio.h>
#include <stdlib.h>

int main() {
    int n;
    scanf("%d", &n);
    
    int *p = (int *) malloc(n * sizeof(int));
    int *p_copy = (int *) malloc(n * sizeof(int));
    
    for (int i = 0; i < n; i++) {
        scanf("%d", &p[i]);
        p_copy[i] = p[i];
    }
    
    for (int i = 0; i < n; i++) {
        for (int j = i + 1; j < n; j++) {
            if (p[i] > p[j]) {
                int temp = p[i];
                p[i] = p[j];
                p[j] = temp;
            }
        }
    }
    
    int count = 0;
    for (int i = 0; i < n; i++) {
        if (p[i] != p_copy[i]) {
            count++;
        }
    }
    
    if (count == 0 || count == 2) {
        printf("YES\n");
    } else {
        printf("NO\n");
    }
    
    free(p);
    free(p_copy);
    
    return 0;
}
// 
